homeassistant/components/wallbox/config_flow.py

```diff
@@ -0,0 +1,114 @@
+"""Config flow for Wallbox integration."""
+import logging
+
+import requests
+import voluptuous as vol
+from wallbox import Wallbox
+
+from homeassistant import config_entries, core, exceptions
+
+from .const import DOMAIN
+
+COMPONENT_DOMAIN = DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        "station": str,
+        "username": str,
+        "password": str,
+    }
+)
+
+
+class PlaceholderHub:
+    """Wallbox Hub class."""
+
+    def __init__(self, station, username, password):
+        """Initialize."""
+        self._station = station
+        self._username = username
+        self._password = password
+
+    def authenticate(self) -> bool:
+        """Authenticate using Wallbox API."""
+        try:
+            wallbox = Wallbox(self._username, self._password)
+            wallbox.authenticate()
+            return True
+        except requests.exceptions.HTTPError as wallbox_connection_error:
+            if wallbox_connection_error.response.status_code == 403:
+                raise InvalidAuth from wallbox_connection_error
+            raise ConnectionError from wallbox_connection_error
+
+    def get_data(self) -> bool:
+        """Get new sensor data for Wallbox component."""
+
+        try:
```

Comment:
I suggest to make a function with the block and a function object as parameter to call e.g. getChargerStatus

You might also be able to centralise the add_executor_job, and thus be free to make many of the methods async.

Reply:
That's a good suggestion, that would allow me to remove some duplicate code and make it more clean. I'm going to write that down as a suggestion for a next release if that's ok with you.

Reply:
We can park "make a function" to a later PR, that is no problem. But your distributed add_executor_job is a concern. I know you need to use them (I do the same in modbus/modbus.py), but please try to keep as close to the 3rd party library as possible and preference is to isolate them in WallboxHub. Doing makes the bigger part of your code clean and nice async.